### PR TITLE
calendar rework

### PR DESCRIPTION
**Pull request by devthehuman**
## What's this PR do?
* Changes the text of the 'add to calendar' button to 'remove from calendar' if the plan exists in the user's calendar, and redirects to the calendar page on click. 

## Pre-Merge TODOs
 - [X] Screenshots Included
 - [X] Tests Added
 - [ ] Updated Documentation
 - [ ] Add follow up issues?
 - [ ] 2 PR reviews completed

## Learning
* n/a

## Where should the reviewer start?
* EventDetails.js

## Any background context you want to provide?
* Pretty straightforward update. 

## Screenshots
* When plan does not exist in the db
screen shot 2017-06-07 at 11 20 09 am

* When plan is added to the db or already exists
screen shot 2017-06-07 at 11 20 29 am

### CODE REVIEW

File: app/components/EventDetails.js [hunk 1]
```diff
@@ -57,10 +61,7 @@ const EventDetails = ({ user, id, name, images, dates, classifications, _embedde
                 onClick={ () => history.replace('/') }>
                 Return
               </button>
-        <button className={ toggleClass() }
-                onClick={ () => addNewPlan() }>
-                Save To Calendar
-              </button>
+        { toggleButton() }
```
Comment (cbandrow) [inline app/components/EventDetails.js]: I dig this component refactoring.

**Comment by JohnBinning** [comment]
When plan is added to the db or already exists - consider changing font color on the remove from calendar button

File: tests/components/EventDetails.test.js [hunk 2]
```diff
@@ -135,11 +135,58 @@ describe('EventDetails', () => {
                                           plans={ newPlans }
                                           history={ browserHistory }/>);
 
-    const saveButton = wrapper.find('saved-plan');
+    const saveButton = wrapper.find('.saved-plan');
 
     saveButton.simulate('click');
 
     expect(mockDelete).toHaveBeenCalledTimes(1);
     expect(mockFetch).toHaveBeenCalledTimes(1);
```
Comment (JohnBinning) [inline tests/components/EventDetails.test.js]: nice work with the separate mock functions

**Comment by devthehuman** [comment]
@tman22, @rrgayhart, @joshuajhun for review

**Pull request merged**